### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by marcransome**
Resolves: `INC0373474`.

### CODE REVIEW

File: groups/storage/locals.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+locals {
+  common_tags = {
+    Environment = var.environment
+    Service     = var.service
+    Repository  = var.repository
```
Comment (Jata2k) [inline groups/storage/locals.tf:5]: We'll need to agree on standards for this. We've just gone with `Terraform` as the provisioning repository elsewhere...

File: groups/storage/main.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "eu-west-2"
```
Comment (Jata2k) [inline groups/storage/main.tf:13]: Is there a reason to hard code this? We've historically used `var.region` elsewhere

File: groups/storage/s3.tf [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  bucket = "${var.service}.${var.environment}.${var.aws_account}.ch.gov.uk"
+  tags   = local.common_tags
+}
+
+resource "aws_s3_bucket_acl" "data" {
+  bucket = aws_s3_bucket.data.id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_versioning" "versioning_example" {
```
Comment (Jata2k) [inline groups/storage/s3.tf:11]: Copy paste typo? `versioning_example `

File: groups/storage/locals.tf (hunk 1 above)
Comment (marcransome) [reply groups/storage/locals.tf:5]: Tags reworked in 3803c3f as discussed.

**Review by Jata2k — APPROVED** [review]

**Pull request merged**